crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3433,6 +3433,23 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_assignment_in_assignment() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_dedented(
+            "/src/a.py",
+            "
+            x = [0]
+            x[0 if (y := 2) else 1] = 1
+            ",
+        )?;
+
+        assert_file_diagnostics(&db, "/src/a.py", &[]);
+
+        Ok(())
+    }
```

Comment:
We should avoid adding new tests to `infer.rs` and instead use our type inference testing framework. https://github.com/astral-sh/ruff/tree/main/crates/red_knot_test

Reply:
Moved this over to an md test